File: lista-de-exercicios/exercicio-3/Jokenpo.test.ts
import { jest } from '@jest/globals';
import { Escolhas } from './Jokenpo.js';
import { Jokenpo } from './Jokenpo';
import { Jogador } from './Jogador';

const pedra = Escolhas.PEDRA;
const papel = Escolhas.PAPEL;
const tesoura = Escolhas.TESOURA;
const jogadorPedra = new Jogador(pedra);
const jogadorPapel = new Jogador(papel);
const jogadorTesoura = new Jogador(tesoura);

describe('Classe Jokenpo', () => {
    test('Pedra ganha de tesoura', () => {
        const jokenpo1 = new Jokenpo(jogadorPedra, jogadorTesoura);
        const jokenpo2 = new Jokenpo(jogadorTesoura, jogadorPedra);

        expect(jokenpo1.jogar()).toBe("Vencedor: Player 1 com PEDRA");
        expect(jokenpo2.jogar()).toBe("Vencedor: Player 2 com PEDRA");
    })

    test('Papel ganha de pedra', () => {
        const jokenpo1 = new Jokenpo(jogadorPapel, jogadorPedra);
        const jokenpo2 = new Jokenpo(jogadorPedra, jogadorPapel);

        expect(jokenpo1.jogar()).toBe("Vencedor: Player 1 com PAPEL");
        expect(jokenpo2.jogar()).toBe("Vencedor: Player 2 com PAPEL");
    })

    test('Tesoura ganha de papel', () => {
        const jokenpo1 = new Jokenpo(jogadorTesoura, jogadorPapel);
        const jokenpo2 = new Jokenpo(jogadorPapel, jogadorTesoura);

        expect(jokenpo1.jogar()).toBe("Vencedor: Player 1 com TESOURA");
        expect(jokenpo2.jogar()).toBe("Vencedor: Player 2 com TESOURA");
    })

    test('Escolhas iguais retornam empate', () => {
        const jokenpoPedra = new Jokenpo(jogadorPedra, jogadorPedra);
        const jokenpoPapel = new Jokenpo(jogadorPapel, jogadorPapel);
        const jokenpoTesoura = new Jokenpo(jogadorTesoura, jogadorTesoura);

        expect(jokenpoPedra.jogar()).toBe('Empate');
        expect(jokenpoPapel.jogar()).toBe('Empate');
        expect(jokenpoTesoura.jogar()).toBe('Empate');
    })
})

File: lista-de-exercicios-java/src/test/java/org/example/exercicio03/JokenpoTest.java
package org.example.exercicio03;

import static org.example.exercicio03.Escolha.PEDRA;
import static org.example.exercicio03.Escolha.PAPEL;
import static org.example.exercicio03.Escolha.TESOURA;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class JokenpoTest {
    Jogador jogadorPedra = new Jogador(PEDRA);
    Jogador jogadorPapel = new Jogador(PAPEL);
    Jogador jogadorTesoura = new Jogador(TESOURA);

    @DisplayName("A escolha pedra deve ganhar de tesoura")
    @Test
    void pedraGanhaTesoura() {
        Jokenpo jokenpo1 = new Jokenpo(jogadorPedra, jogadorTesoura);
        Jokenpo jokenpo2 = new Jokenpo(jogadorTesoura, jogadorPedra);

        assertTrue(jokenpo1.jogar().equals("Vencedor: Player 1 com PEDRA"));
        assertTrue(jokenpo2.jogar().equals("Vencedor: Player 2 com PEDRA"));
    }

    @DisplayName("A escolha papel deve ganhar de pedra")
    @Test
    void papelGanhaPedra() {
        Jokenpo jokenpo1 = new Jokenpo(jogadorPapel, jogadorPedra);
        Jokenpo jokenpo2 = new Jokenpo(jogadorPedra, jogadorPapel);

        assertTrue(jokenpo1.jogar().equals("Vencedor: Player 1 com PAPEL"));
        assertTrue(jokenpo2.jogar().equals("Vencedor: Player 2 com PAPEL"));
    }

    @DisplayName("A escolha tesoura deve ganhar de papel")
    @Test
    void tesouraGanhaPapel() {
        Jokenpo jokenpo1 = new Jokenpo(jogadorTesoura, jogadorPapel);
        Jokenpo jokenpo2 = new Jokenpo(jogadorPapel, jogadorTesoura);

        assertTrue(jokenpo1.jogar().equals("Vencedor: Player 1 com TESOURA"));
        assertTrue(jokenpo2.jogar().equals("Vencedor: Player 2 com TESOURA"));
    }
}